app/views/questionnaire.py

```diff
@@ -104,12 +101,13 @@ def before_post_submission_request():
     logger.bind(eq_id=values['eq_id'], form_type=values['form_type'])
     logger.info('questionnaire request', method=request.method, url_path=request.full_path)
 
-    _check_same_survey(url_eq_id=values['eq_id'],
-                       url_form_type=values['form_type'],
-                       url_collection_id='',
-                       session_eq_id=session_data.eq_id,
-                       session_form_type=session_data.form_type,
-                       session_collection_id='')
+    metadata_from_session_data = {
+        'tx_id': session_data.tx_id,
+        'eq_id': session_data.eq_id,
+        'form_type': session_data.form_type,
+    }
+    if check_multiple_survey(metadata_from_session_data, values):
```

Comment:
We probably should also add this check in feedback?.
https://github.com/ONSdigital/eq-survey-runner/blob/master/app/views/feedback.py#L29-L36

At the moment, if i try POST a feedback on a previous survey, it says 'Thanks for your feedback', even though it was rejected by CSRF.

Reply:
So the code wouldn't get hit here as it handles the CSRFError first. So it is already hitting the new `csrf_error` handler in `errors.py`. Since it doesn't have `request.view_args` values it returns false from `check_multiple_survey`. It is therefore returning the `session-expired.html` but that doesn't filter to the page as a whole. If i'm honest I don't know how to easily fix this.